website/server/controllers/api-v3/groups.js

```diff
@@ -303,6 +305,18 @@ api.joinGroup = {
 
     let promises = [group.save(), user.save()];
 
+    if (inviter) {
+      inviter = await User.findById(inviter).select('notifications preferences.language').exec();
+
+      let data = {
+        headerText: common.i18n.t('invitationAcceptedHeader', inviter.preferences.language),
+        bodyText: common.i18n.t('invitationAcceptedBody', {groupName: group.name, username: user.auth.local.username},
+          inviter.preferences.language),
+      };
+      inviter.addNotification('GROUP_INVITE_ACCEPTED', data);
+      promises.push(inviter.save());
+    }
+
     if (group.type === 'party' && inviter) {
       promises.push(User.update({_id: inviter}, {$inc: {'items.quests.basilist': 1}}).exec()); // Reward inviter
```

Comment:
since we load the inviter, it's better to do the modification on the inviter object and then `save` instead of a separate update (you'll need to load `items.quests.basilist` other than `notifications preferences.language `